pkg/reconcile/pipeline/infinispan/handler/provision/service_monitor.go

```diff
@@ -68,6 +68,7 @@ func ServiceMonitor(i *iv1.Infinispan, ctx pipeline.Context) {
 			NamespaceSelector: monitoringv1.NamespaceSelector{
 				MatchNames: []string{i.Namespace},
 			},
+			TargetLabels: i.ServiceMonitorTargetLabels(),
```

Comment:
You should be able to remove this as the `mutateFn` will always be applied, even on initial create.

Reply:
removed the line